src/main/java/net/mcreator/ui/dialogs/UpdatePluginDialog.java

```diff
@@ -41,24 +41,37 @@ public static void showPluginUpdateDialogIfUpdatesExist(Window parent) {
 
 			pan.add("North", L10N.label("dialog.plugin_update_notify.message"));
 			pan.add("Center", new JScrollPane(PanelUtils.pullElementUp(plugins)));
+			pan.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
 			pan.setPreferredSize(new Dimension(560, 250));
 
 			for (PluginUpdateInfo pluginUpdateInfo : PluginLoader.INSTANCE.getPluginUpdates()) {
-				JLabel label = L10N.label("dialog.plugin_update_notify.version_message",
+				StringBuilder usb = new StringBuilder(L10N.t("dialog.plugin_update_notify.version_message",
 						pluginUpdateInfo.plugin().getInfo().getName(), pluginUpdateInfo.plugin().getInfo().getVersion(),
-						pluginUpdateInfo.newVersion());
+						pluginUpdateInfo.newVersion()));
+				if (pluginUpdateInfo.recentChanges() != null) {
+					usb.append("<br>").append(L10N.t("dialog.plugin_update_notify.changes_message")).append("<ul>");
+					for (String change : pluginUpdateInfo.recentChanges())
+						usb.append("<li>").append(change).append("</li>");
+				}
 
+				JLabel label = new JLabel(usb.toString());
 				JButton update = L10N.button("dialog.plugin_update_notify.update");
 				update.addActionListener(e -> DesktopUtils.browseSafe(
 						MCreatorApplication.SERVER_DOMAIN + "/node/" + pluginUpdateInfo.plugin().getInfo()
 								.getPluginPageID()));
-
-				plugins.add(PanelUtils.westAndEastElement(label, update));
+				plugins.add(PanelUtils.westAndEastElement(label, PanelUtils.join(update)));
 			}
 
-			JOptionPane.showOptionDialog(parent, pan, L10N.t("dialog.plugin_update_notify.update_title"),
-					JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE, null,
-					new Object[] { L10N.t("dialog.plugin_update_notify.close") }, "");
+			MCreatorDialog window = new MCreatorDialog(null, L10N.t("dialog.plugin_update_notify.update_title"));
+			window.setSize(720, 300);
+			window.setLocationRelativeTo(parent);
+			window.setModal(true);
+
+			JButton close = L10N.button("dialog.plugin_update_notify.close");
+			close.addActionListener(e -> window.setVisible(false));
+
+			window.getContentPane().add("Center", PanelUtils.centerAndSouthElement(pan, PanelUtils.join(close)));

```

Comment:
`window.getContentPane().add` -> `window.add`

window could also be renamed to dialog as window is something else in swing